src/xeus_sqlite_interpreter.cpp

```diff
@@ -149,47 +154,54 @@ void interpreter::delete_db()
 
     if(std::remove(m_db_path.c_str()) != 0)
     {
-        publish_stream("stderr", "Error deleting file.\n");
-    }
-    else
-    {
-        m_bd_is_loaded = false;
-        publish_stream("stdout", "File successfully deleted.\n");
+        std::string error = "Error deleting file.";
+        m_traceback.push_back("Error: " + error);
+        publish_execution_error("Error", error, m_traceback);
+        m_traceback.clear();
     }
 }
 
-void interpreter::table_exists(const std::string table_name)
+void interpreter::table_exists(int execution_counter, const std::string table_name)
 {
     if (m_db->SQLite::Database::tableExists(table_name.c_str()))
     {
-        publish_stream("stdout", "The table " + table_name + " exists.\n");
+        nl::json pub_data;
+        pub_data["text/plain"] = "The table " + table_name + " exists.";
+        publish_execution_result(execution_counter, std::move(pub_data), nl::json::object());
     }
     else
     {
-        publish_stream("stdout", "The table " + table_name + " doesn't exist.\n");
+        nl::json pub_data;
+        pub_data["text/plain"] = "The table " + table_name + " doesn't exist.";
+        publish_execution_result(execution_counter, std::move(pub_data), nl::json::object());
     }
 }
 
-void interpreter::is_unencrypted()
+void interpreter::is_unencrypted(int execution_counter)
 {
     if (SQLite::Database::isUnencrypted(m_db_path))
     {
-        publish_stream("stdout", "The database is unencrypted.\n");
+        nl::json pub_data;
+        pub_data["text/plain"] = "The database is unencrypted.";
+        publish_execution_result(execution_counter, std::move(pub_data), nl::json::object());
     }
     else
     {
-        publish_stream("stdout", "The database is encrypted.\n");
+        nl::json pub_data;
+        pub_data["text/plain"] = "The database is encrypted.";
+        publish_execution_result(execution_counter, std::move(pub_data), nl::json::object());
     }
 }
 
-void interpreter::get_header_info()
+void interpreter::get_header_info(int execution_counter)
```

Comment:
Instead of taking the execution count as input, what if you change it so that it returns the publication data?

`get_header_info` sounds a lot like it would return something, instead of void, and it's a bit weird in the design to take the execution count as input.

Reply:
I guess it's the same for all methods that you changed.

Reply:
Yeah, that's a good idea.
Haven't thought of that. Will apply later today. Thanks Martin.